client/src/PagesBody/ContestsPage/ContestsPage.js

```diff
@@ -33,33 +34,46 @@ const ContestsPage = () => {
 
   useEffect(() => {
     async function fetchContestsData() {
-      const response = await fetch(
-        '/api/contests/card/data',
-        requestOptionsGET,
-      );
-      const result = await response.json();
+      let result = null;
+      if (
+        state.selectedRole === ROLE_NAME.PARTICIPANT &&
+        locationPath.state &&
+        locationPath.state.contestContent !== 'future'
+      ) {
+        result = await getContestsCards(ROLE_NAME.PARTICIPANT);
+      } else if (state.selectedRole === ROLE_NAME.MANAGER) {
+        result = await getContestsCards(ROLE_NAME.MANAGER);
+      } else if (state.selectedRole === ROLE_NAME.STAFF) {
+        result = await getContestsCards(ROLE_NAME.STAFF);
+      } else {
+        result = await getContestsCards();
+      }
+      if (Array.isArray(result)) {
+        result = removeNullsFromArray(result);
+      }
       rawDataFromDB.current = result;
 
-      if (
+      if (rawDataFromDB.current === null) {
+        setIsPending(false);
+      } else if (
+        locationPath.state &&
```

Comment:
locationPath.state seems to be everywhere also, perhaps we can take it outside of it so its not repeated? and then use swithc just for the contestContent?